src/System Application/App/Cryptography Management/src/RSA.Codeunit.al

```diff
@@ -0,0 +1,169 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.Security.Encryption;
+
+/// <summary>
+/// Performs asymmetric encryption and digital signature using the implementation of the RSA class.
+/// </summary>
+codeunit 5 "RSA"
+{
+    Access = Public;
+    InherentEntitlements = X;
+    InherentPermissions = X;
+
+    var
+        [NonDebuggable]
+        RSAImpl: Codeunit "RSA Impl.";
+
+    /// <summary>
+    /// Initializes a new instance of RSA with the specified key size.
+    /// </summary>
+    /// <param name="KeySize">The size of the key in bits.</param>
+    procedure InitializeRSA(KeySize: Integer)
+    begin
+        RSAImpl.InitializeRSA(KeySize);
+    end;
+
+#if not CLEAN24
+    /// <summary>
+    /// Creates and returns an XML string containing the key of the current RSA object.
+    /// </summary>
+    /// <param name="IncludePrivateParameters">true to include a public and private RSA key; false to include only the public key.</param>
+    /// <returns>An XML string containing the key of the current RSA object.</returns>
+    [NonDebuggable]
+    [Obsolete('Use ToXmlString with SecretText data type for XmlString.', '24.0')]
```

Comment:
Remove all the obsoleted functions. We shouldn't be adding them as they're insecure.
There is no implementing of an interface on this facade, so we can remove them.